lib/constructs/interface.js

```diff
@@ -130,6 +132,52 @@ Interface.prototype.generateExport = function () {
     exposers.push(keys[i] + ": { " + exposedOnObj.join(", ") + " }");
   }
 
+  if (this.iterable) {
+    this.str += `\nconst IteratorPrototype = Object.create(utils.IteratorPrototype, {
+  next: {
+    value: function next() {
+      const internal = this[utils.iterInternalSymbol];
+      const target = internal.target;
+      const kind = internal.kind;
+      const index = internal.index;
+      const values = Array.from(target[impl]);
```

Comment:
Oh, nope, it works fine, as long as you snapshot *each time* next() is called. This implementation is inefficient, but we can tidy that up later.